client/jetpack-cloud/sections/partner-portal/utils.ts

```diff
@@ -12,16 +10,39 @@ export const STATE_REVOKED = 'revoked';
  *
  * @param {string | null} attachedAt Date the license was attached on, if any.
  * @param {string | null} revokedAt Date the license was revoked on, if any.
- * @returns {string} State matching one of the STATE_* constant values.
+ * @returns {LicenseStates} State matching one of the `LicenseStates` values.
  */
-export function getLicenseState( attachedAt: string | null, revokedAt: string | null ): string {
+export function getLicenseState(
+	attachedAt: string | null,
+	revokedAt: string | null
+): LicenseStates {
 	if ( revokedAt ) {
-		return STATE_REVOKED;
+		return LicenseStates.Revoked;
 	}
 
 	if ( attachedAt ) {
-		return STATE_ATTACHED;
+		return LicenseStates.Attached;
 	}
 
-	return STATE_DETACHED;
+	return LicenseStates.Detached;
+}
+/**
+ * Get the state of a license based on a string.
+ * If the given string is not a valid LicenseState, it returns `LicenseStates.All`
+ *
+ * @param {(LicenseStates|string)?} licenseFilter The filter string matching the `state` param in the URL
+ * @returns {LicenseStates} State matching one of LicenseStates values.
+ */
+export function getLicenseStateByQueryParamValue(
+	licenseFilter?: LicenseStates | string
+): LicenseStates {
+	switch ( licenseFilter ) {
+		case LicenseStates.All:
+		case LicenseStates.Attached:
+		case LicenseStates.Detached:
+		case LicenseStates.Revoked:
+			return licenseFilter;
+		default:
+			return LicenseStates.All;
+	}
```

Comment:
What was the problem with the previous includes() approach? This change would require devs to be aware of this function and keep it in sync with the enum if we introduce new options in the future.
IMO something like this would be better:
```ts
export function toLicenseFilter( value: string ): LicenseFilter {
	return Object.values( LicenseFilter ).includes( value as LicenseFilter )
		? ( value as LicenseFilter )
		: LicenseFilter.All;
}
```

PS:
I was under the impression TS supported something like `toEnum<T extends enum>( value: string | number, defaultValue: T ) T {}` which would have been even better but unfortunately it doesn't :(

Reply:
>What was the problem with the previous includes() approach? 

No problem with the other approach, I was just thinking of what would be easier to understand at a glance. I also thought about the dev being aware of that function, but a simple unit test would help there, considering there are many other places where we have to change when adding/removing a LicenseFilter and, realistically, the LicenseFilter would probably not change (or at least not so often).